src/track/beatmap.cpp

```diff
@@ -225,23 +234,29 @@ BeatsPointer BeatMap::makeBeatMap(
 BeatsPointer BeatMap::makeBeatMap(
         audio::SampleRate sampleRate,
         const QString& subVersion,
-        const QVector<double>& beats) {
+        const QVector<mixxx::audio::FramePos>& beats) {
     BeatList beatList;
 
-    double previous_beatpos = -1;
-    Beat beat;
+    mixxx::audio::FramePos previousBeatPos = mixxx::audio::kInvalidFramePos;
 
-    foreach (double beatpos, beats) {
-        // beatpos is in frames. Do not accept fractional frames.
-        beatpos = floor(beatpos);
-        if (beatpos <= previous_beatpos || beatpos < 0) {
-            qDebug() << "BeatMap::createFromVector: beats not in increasing order or negative";
-            qDebug() << "discarding beat " << beatpos;
-        } else {
-            beat.set_frame_position(static_cast<google::protobuf::int32>(beatpos));
-            beatList.append(beat);
-            previous_beatpos = beatpos;
+    for (const mixxx::audio::FramePos originalBeatPos : beats) {
```

Comment:
We might need to remove the `const` qualifier to avoid this warning even though it is the most restrictive option and usually what is intended. Other range-based for loops that don't use const don't trigger this warning.